## REQUEST_CHANGES — entrega não atende aos critérios de aceite da issue #432

Verifiquei o estado real rodando `python3 -m pytest deile/tests/ -q --timeout=120 -p no:randomly` em HEAD `70af563`. Resultado: **18 failed, 6094 passed, 41 skipped**. A issue exige **exatamente 4 failed** (AC #6 — só os 4 pré-existentes de #433).

### 1. AC #5 violado — `test_renderer_task_awaited_before_stdout_restore` continua FALHANDO no run completo

```
FAILED deile/tests/orchestration/test_subagent_orchestrator.py::test_renderer_task_awaited_before_stdout_restore
       - assert <deile.orchestration.subagents._capture.SwitchableStream object at 0...
```

O fixture `_guard_sys_stdio` snapshota `sys.stdout` no setup da função. Se um teste anterior já deixou `sys.stdout` apontando para um `SwitchableStream` quando o setup deste teste roda, o fixture snapshota o valor já poluído e "restaura" para o mesmo valor poluído ao final — não restaura ao stdout original. A causa-raiz não foi atacada.

A própria issue mandou, no AC #4.3, `_assert_stdio_unchanged` que **FALHA o teste** se stdio mudou (sem `monkeypatch.setattr`), justamente pra forçar a aparição do leaker em vez de mascarar. O PR implementou silent restore.

### 2. AC #6 violado — 11 NOVAS falhas ordering-dependent INTRODUZIDAS pelo próprio fix

Não são pré-existentes — a issue documenta baseline de 13 failed (9 ordering + 4 de #433). Atual: 18. Cada uma das 11 novas passa em isolamento:

```
test_dispatch_resolver_settings.py (7 falhas, todas verdes em isolamento):
  test_resolver_settings_per_stage_wins_over_env_global         assert 'deile-worker' == 'claude-worker'
  test_resolver_settings_global_wins_over_default               assert 'deile-worker' == 'claude-worker'
  test_resolver_all_stages_via_settings_per_stage
  test_resolver_stages_are_independent_per_settings
  test_resolver_invalid_settings_per_stage_falls_through_with_warning
  test_resolver_invalid_settings_global_falls_through_with_warning
  test_resolver_invalid_dispatcher_value_falls_through_with_warning
  test_resolver_settings_per_stage_canonicalizes_legacy_aliases
  test_resolver_full_precedence_chain_all_layers

test_deile_md_loader.py (4 falhas, todas verdes em isolamento):
  test_merged_prompt_disabled_via_settings_returns_empty
  test_settings_override_user_path                              assert None is not None
  test_settings_override_cwd_filename                           assert None is not None
  test_large_file_is_truncated                                  assert 5000 <= 100
```

Provei em isolamento: `pytest deile/tests/orchestration/pipeline/test_dispatch_resolver_settings.py deile/tests/test_deile_md_loader.py` → **43/43 passed**. Os mesmos testes falham com o run completo.

Hipótese: o `_snapshot_os_environ` + `_clean_logging_handlers` interagem com testes que mexem em settings/env via outros caminhos (registro hot-reload de skills, configuração de modelo, settings overrides) e produzem nova classe de leaks que o conftest cria em vez de prevenir. Sem investigação da causa-raiz por teste leaker (AC #7), trocamos 1 falha por 11.

### 3. AC #7 violado — body do PR não identifica o leaker por teste

A issue exige explicitamente: *"o PR de fix DEVE incluir, no body, qual teste anterior é o leaker para cada um dos 9 casos — não apenas mascarar com snapshot/restore"*. Aceita-se "não foi possível identificar via bisseção" SOMENTE para os 7 de `TestTickSummary`.

O body do PR descreve apenas as fixtures adicionadas. Nenhum leaker é nomeado para nenhum dos 9 casos. Mascarar sem identificar a causa é exatamente o que a issue proibiu — e o resultado prático (11 novos leaks introduzidos) confirma o porquê.

### 4. Seção "Testes" da issue completamente ignorada

AC mandam 5 artefatos de teste novos. **Zero foram entregues**:

```
deile/tests/test_conftest_isolation.py                    FALTANDO
deile/tests/test_ordering_smoke.py                        FALTANDO
test_warns_even_after_environ_leak (regression)           FALTANDO
test_tick_summary_records_count_isolated_from_external_handlers  FALTANDO
CI job 5x random seed                                     FALTANDO
```

Sem esses testes meta, regressões futuras no próprio mecanismo de isolamento passarão batidas — exatamente o cenário pra qual a issue foi aberta.

### 5. Naming/comportamento divergem do AC #4

| AC #4 prescrito | PR entregou |
|---|---|
| `_snapshot_logging_handlers` | `_clean_logging_handlers` (funcional, ok) |
| `_assert_stdio_unchanged` — **FALHA** o teste se stdio mudou | `_guard_sys_stdio` — **silent restore** (mascarou, e por isso AC #5 falhou) |

### 6. Body do PR contém afirmação incompleta

> "Full orchestration test suite regression check: pytest deile/tests/orchestration/pipeline/ deile/tests/orchestration/test_subagent_orchestrator.py -p no:cov -q → 1066 passed, 2 skipped"

Verdadeiro pro subset que foi rodado, mas a regressão real (11 testes quebrados, incluindo 4 em `deile/tests/test_deile_md_loader.py`, fora do diretório `orchestration/`) não foi exercida. O gate de CI real (`pytest deile/tests/ -q`, com `--cov-fail-under=80`) é o portão obrigatório segundo `CLAUDE.md` e Decisão #45/#46 — esse portão está vermelho.

---

### O que precisa pra unblockar

1. Investigar e identificar o leaker concreto pra cada um dos 9 testes (ou justificar bisseção falha SOMENTE pros 7 de `TestTickSummary`), e documentar no body. Para `test_renderer_task_awaited_before_stdout_restore` em particular: trocar `_guard_sys_stdio` por um fixture que **falha o teste** quando stdio muda sem `monkeypatch` — isso vai cuspir o leaker.
2. Investigar e corrigir a causa-raiz das **11 NOVAS regressões ordering-dependent** introduzidas pelas fixtures (`test_dispatch_resolver_settings.py` x7, `test_deile_md_loader.py` x4). Snapshot/restore agressivo está interagindo com outro estado global (provavelmente settings/skills registry hot-reload). 
3. Entregar os 5 artefatos de teste mandados pela seção "Testes" da issue.
4. Confirmar AC #6 com run completo: `pytest deile/tests/ -q --timeout=120` produzindo exatamente 4 failed (todos em #433).
5. Confirmar AC #2 com 5 seeds aleatórios distintos pós-`pytest-randomly`.

A direção arquitetural (snapshot/restore via autouse + `pytest-randomly`) está correta — só precisa ser executada até o fim, não em meia ponte.

--- a/deile/tests/conftest.py
+++ b/deile/tests/conftest.py
@@ -47,10 +47,16 @@ def _clean_logging_handlers():
     tests using caplog.at_level(..., logger=name) receive 0 records even when
     the code does emit — this is the root cause of the 7 TestTickSummary and
     1 test_warns_when_no_tokens ordering failures (issue #432).
+
+    Also restores logging.Manager.disable: deile/cli.py calls logging.disable()
+    to suppress output during CLI runs; without this restore, all subsequent
+    logging.info() / logging.warning() calls return False from isEnabledFor()
+    and are silently dropped — causing deile/tests/log_mgmt/ failures.
     """
     root = logging.root
     root_handlers_before = root.handlers[:]
     root_level_before = root.level
+    manager_disable_before = root.manager.disable
 
     mgr = logging.Logger.manager
     snapshot: dict = {}
@@ -63,6 +69,9 @@ def _clean_logging_handlers():
             }
 
     yield
+
+    if root.manager.disable != manager_disable_before:
+        logging.disable(manager_disable_before)
 
     root.handlers[:] = root_handlers_before
     root.level = root_level_before

